test/widgets/message_list_test.dart

```diff
@@ -605,6 +606,86 @@ void main() {
           expectedMessage: 'NetworkException: Oops (ClientException: Oops)');
       });
     });
+
+    group('Update Narrow on message move', () {
+      final channel = eg.stream(name: 'move test stream');
+      final message = eg.streamMessage(stream: channel, content: 'Message to move');
+      final narrow = TopicNarrow.ofMessage(message);
+
+      void prepareGetMessageResponse(List<Message> messages) {
+        connection.prepare(json: newestResult(
+          foundOldest: false, messages: messages).toJson());
+      }
+
+      void handleMessageMoveEvent(List<StreamMessage> messages, String newTopic) {
+        store.handleEvent(eg.updateMessageEventMoveFrom(
+          origMessages: messages,
+          newTopic: newTopic,
+          propagateMode: PropagateMode.changeAll));
+      }
+
+      testWidgets('compose box send message after move', (WidgetTester tester) async {
+        await setupMessageListPage(tester, narrow: narrow, messages: [message], streams: [channel]);
+
+        final channelContentInputFinder = find.descendant(
+          of: find.byType(ComposeAutocomplete),
+          matching: find.byType(TextField));
+
+        await tester.enterText(channelContentInputFinder, 'Some text');
+        prepareGetMessageResponse([message]);
+        handleMessageMoveEvent([message], 'new topic');
+        await tester.pump(const Duration(seconds: 1));
+        check(tester.widget<TextField>(channelContentInputFinder))
+          ..decoration.isNotNull().hintText.equals('Message #${channel.name} > new topic')
+          ..controller.isNotNull().text.equals('Some text');
+
+        connection.prepare(json: SendMessageResult(id: 1).toJson());
+        await tester.tap(find.byIcon(Icons.send));
+        await tester.pump();
+        check(connection.lastRequest).isA<http.Request>()
+          ..method.equals('POST')
+          ..url.path.equals('/api/v1/messages')
+          ..bodyFields.deepEquals({
+            'type': 'stream',
+            'to': '${message.streamId}',
```

Comment:
Similarly, because the message gets mutated this should refer to the stream/channel object and not the message.